static/src/javascripts/projects/commercial/modules/dfp/prepare-a9.ts

```diff
@@ -19,24 +20,20 @@ const setupA9 = (): Promise<void | boolean> => {
 		return Promise.resolve();
 	}
 
-	let moduleLoadResult = Promise.resolve();
 	if (
 		shouldIncludeOnlyA9 ||
 		(dfpEnv.hbImpl.a9 &&
 			commercialFeatures.dfpAdvertising &&
 			!commercialFeatures.adFree &&
 			!window.guardian.config.page.hasPageSkin)
 	) {
-		moduleLoadResult = import(
-			/* webpackChunkName: "a9" */ '../../../../lib/a9-apstag.js'
-		).then(() => {
-			a9.initialise();
+		// Load a9 third party stub
+		a9Apstag();
```

Comment:
Is the chunking strategy affected? Are we sure this code is not eagerly loaded?

Reply:
Oops I should've re-marked this as draft. I'll update the description when this is ready, but the intention is that we're changing chunking so these scripts are included in the main bundle (they're only a couple of hundred bytes parsed size each). To defer the scripts execution until the point at which they're dynamically imported (so they don't have side effects until we obtain consent etc), we've wrapped them in `() => { ... }` functions we can call.

Reply:
Yes, I had a peek at `commercial-core` and realised they were quite small.

I’m still confused why wrapping in an IIFE adds anything over a function, but I’ll let you be the judge of that.

In my understanding, the following three are identical:

```ts
function doThing1() {
  console.log("thing");
}

const doThing2 = () => {
  console.log("thing");
}

const doThing3 = () => {
  (function() { console.log("thing") })();
}
```

Reply:
So this was the original setup:
```typescript
// stub.js
cosole.log('Do some work');
```

```typescript
// prepare-thing.ts
if (conditionsAreMet) {
  // Top-level console.log won't occur until conditions are met
  await import('./stub.js');
}
```

Which I'm proposing to replace with:
```typescript
// stub.js
const stub = () => {
  console.log('Do some work');
};
```

```typescript
// prepare-thing.ts
import { stub } from './stub.js';

if (conditionsAreMet) {
  // Top-level console.log won't occur until conditions are met
  stub();
}
```

I think the outcome is the same, just with a different chunking strategy.